resolwe_bio/views.py

```diff
@@ -10,28 +10,33 @@
 from resolwe.flow.views import CollectionViewSet
 from .filters import SampleFilter
 from .models import Sample
-from .serializers import SampleSerializer
+from .serializers import PresampleSerializer, SampleSerializer
 
 
-class SampleViewSet(CollectionViewSet):
+class PresampleViewSet(CollectionViewSet):
     filter_class = SampleFilter
-    serializer_class = SampleSerializer
+    serializer_class = PresampleSerializer
 
-    def get_queryset(self):
-        queryset = Sample.objects.annotate(
-            latest_date=Max('data__created')
-        ).prefetch_related('descriptor_schema')
+    queryset = Sample.objects.annotate(
+        latest_date=Max('data__created')
+    ).prefetch_related(
+        'descriptor_schema'
+    ).filter(
+        presample=True
+    ).order_by('-latest_date')
 
-        # Return annotated samples only (default for GET requests)
-        if not self.request.data:
-            annotated = self.request.query_params.get('annotated', None)
 
-            if annotated == "0":
-                queryset = queryset.filter(~Q(descriptor__has_key='geo') | ~Q(descriptor__geo__has_key='annotator'))
-            else:
-                queryset = queryset.filter(Q(descriptor__has_key='geo') & Q(descriptor__geo__has_key='annotator'))
+class SampleViewSet(CollectionViewSet):
+    filter_class = SampleFilter
+    serializer_class = SampleSerializer
 
-        return queryset.order_by('-latest_date')
+    queryset = Sample.objects.annotate(
+        latest_date=Max('data__created')
```

Comment:
We wish to sort sample by `data__modified` by default. The presamples should be sorted by `data__created` as above.